apps/feedback/models/feedback.go

```diff
@@ -1,30 +1,25 @@
 package models
 
 import (
-	"github.com/iReflect/reflect-app/db/models/fields"
 	"github.com/jinzhu/gorm"
-)
-
-type Category struct {
-	gorm.Model
-	Title    string `gorm:"type:varchar(100);unique_index"`
-	Archived bool
-	Weight   int
-	Items    []Item
-}
+	"time"
 
-type ItemType struct {
-	gorm.Model
-	Title string
-}
+	userModels "github.com/iReflect/reflect-app/apps/user/models"
+)
 
-type Item struct {
+type Feedback struct {
 	gorm.Model
-	Title      string
-	Weight     int
-	CategoryID uint         `gorm:"index"`
-	ItemTypeID uint         `gorm:"index"`
-	Data       fields.JSONB `type: jsonb not null default '{}'::jsonb`
-	Category   Category
-	ItemType   ItemType
+	FeedbackForm   FeedbackForm    `gorm:"ForeignKey:FeedbackFormID; AssociationForeignKey:ID"`
+	FeedbackFormID uint            `gorm:"not null"`
+	ForUser        userModels.User `gorm:"ForeignKey:ForUserID; AssociationForeignKey:ID"`
+	ForUserID      uint
+	ByUser         userModels.User `gorm:"ForeignKey:ByUserID; AssociationForeignKey:ID"`
+	ByUserID       uint            `gorm:"not null"`
+	Team           userModels.Team `gorm:"ForeignKey:TeamID; AssociationForeignKey:ID"`
+	TeamID         uint            `gorm:"not null"`
+	Status         int8            `gorm:"default:0; not null"`
+	SubmittedAt    time.Time
```

Comment:
All these time fields are NULLABLE or not ?

Reply:
These are nullable fields

Reply:
As discussed, we need to create separate constraints on own as no support at gorm level. Will be tackling later